### PR TITLE
Optimize blueprint updater logic and strengthen the test suite

### PR DESCRIPTION
**Pull request by luuquangvu**
## Summary by Sourcery

Harden compatibility validation, blueprint backup handling, and dependency checks while simplifying the test suite and reporting.

New Features:
- Validate and normalize compatibility matrix entries with structured typing and stricter version label checking.
- Probe compatibility test virtualenvs for missing pytest dependencies and ensure they are installed before running tests.
- Add timeout handling and explicit notices for dependency sync and update checks in the validation pipeline.

Bug Fixes:
- Prevent crashes by treating invalid compatibility matrix content and non-string version labels as validation errors rather than assumptions.
- Fix blueprint import conflict messages to reference relative paths and ensure HTTP sessions are always constructed with up-to-date client options.
- Ensure temporary files created during blueprint save and restore operations are cleaned up on both success and failure paths to avoid filesystem leaks.
- Handle filesystem and directory scanning errors gracefully during backup rotation, ignoring malformed backup filenames and limiting stale backup cleanup to valid backups.
- Improve error reporting when compatibility tests or validation steps fail due to subprocess timeouts or validation errors, including labeling matrix row results.

Enhancements:
- Refine backup rotation logic to delete excess backups by scanning the directory rather than relying on a single stale path.
- Consolidate entity lookup helpers for blueprints by renaming and simplifying internal methods used in risk detection and update flows.
- Centralize pytest arguments for compatibility runs and improve result reporting by including matrix row indices in summaries.
- Remove obsolete benchmark configuration and tests to streamline the default test suite and pytest markers.
- Strengthen CI validation of the compatibility matrix file using jq to enforce structure and version format constraints.

Tests:
- Expand coordinator tests to cover backup rotation edge cases, temporary file cleanup behavior, and restore flows under both success and error conditions.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved blueprint updater backup rotation/restore reliability using temp-file staging, safer cleanup on failures, and stronger error handling.
  * Refined import-path conflict messaging and improved breaking-change detection entity lookup.
* **New Features**
  * Compatibility-matrix validation now fails earlier in CI with stricter structured checks.
  * Validation pipeline now enforces timeouts with clearer timeout and dependency-check notices.
* **Chores**
  * Removed benchmark suite/tooling and simplified benchmark collection behavior.
  * Added repository AI agent guidance via `AGENTS.md`.

### CODE REVIEW

File: tools/validate_compatibility.py [hunk 1]
```diff
@@ -187,6 +223,47 @@ def _overrides_file(ha_ver: str) -> Generator[str]:
...
+        "print(json.dumps(versions))\n"
+    )
+    try:
+        result = subprocess.run(
+            [str(python_bin), "-c", code],
+            check=True,
+            capture_output=True,
+            text=True,
+            cwd=_REPO_ROOT,
+        )
```
[the remark this answers is not in the document]
Comment (luuquangvu) [reply tools/validate_compatibility.py:275]: All string values are hardcoded within the script without any external imports, so this warning is invalid.

**Comment by luuquangvu** [comment]
@sourcery-ai review
